astroid/inference.py

```diff
@@ -130,7 +131,10 @@ def infer_map(self, context=None):
         yield new_seq
```

Comment:
L129 shows that the current typing of `nodes.Dict.items` is indeed incorrect as I initially feared 😓 

`helpers.safe_infer` returns `InferenceContext | None`. In `_infer_map` we make sure to only pass `InferenceResult` to `_update_with_replacement` but that also means that the returning dictionaries are made up of `InferenceResult` and not `NodeNG`.
This also seems correct as `Instance` should be in `.items` for:
```python
class A: ...
b = {1: A()}
```

Reply:
I created https://github.com/PyCQA/astroid/pull/1636 to show how that affects the inference and return types here.

Reply:
I typed `Dict.items` mainly based of the `TreeRebuilder` which explains why I only expected `dict[NodeNG, NodeNG]`. You're right though that it should be `dict[InferenceResult, InferenceResult]` instead.

Reply:
Let's update `Dict.items` here too, then.